### PR TITLE
[Feat] Initial Implementation of Flowbroker-UI

### PR DESCRIPTION
**Pull request by cfrancisco**
This PR is the first implementation of Flowbroker-UI, a UI for Flowbroker service based in NodeRed.

### CODE REVIEW

File: app/dojotModule/utils.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+
+const storageForFlows = {};
+
+/**
+* Converting NodeRed's Flows to Dojot's Flows
+*
+* @param {Array} flows NodeRed's Flows
+*/
+const castFlowsToDojot = (flows) => {
+  const flowsElements = JSON.parse(JSON.stringify(storageForFlows));
```
Comment (luissajunior) [inline app/dojotModule/utils.js]: Is really necessary stringfy and parse since the storageForFlows is an object

File: app/dojotModule/utils.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+/**
+* Converting Dojot's Flows to NodeRed's Flows
+*
+* @param {Array} flows Dojot's Flows
+*/
+const castDojotToFlows = (flows) => {
+  const myFlows = [];
+  flows.forEach((flow) => {
+    const flowName = flow.name;
+    const flowList = JSON.parse(JSON.stringify(flow.flow));
```
Comment (luissajunior) [inline app/dojotModule/utils.js]: Same here

File: red.js [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+  }
+  if (parsedArgs.userDir) {
+    settings.userDir = parsedArgs.userDir;
+  }
+
+  try {
+    logger.info("Initializing Node-RED.");
+    RED.init(server, settings);
+  } catch (err) {
+    logger.error("Failed to start server:");
```
Comment (luissajunior) [inline red.js:312]: If RED not init maybe you can put  stateManager.signalNotReady("server");

File: app/dojotModule/utils.js (hunk 1 above)
Comment (cfrancisco) [reply app/dojotModule/utils.js]: This is a quick way of cloning an object.

File: red.js (hunk 3 above)
Comment (cfrancisco) [reply red.js:312]: Good idea! I'll update it.

File: red.js [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+    if (settings.httpAdminRoot !== false) {
+      listenPath += settings.httpAdminRoot;
+    } else if (settings.httpStatic) {
+      listenPath += "/";
+    }
+    return listenPath;
+  }
+
+  RED.start().then(() => {
+    if (settings.httpAdminRoot !== false
```
[the remark this answers is not in the document]
Comment (cfrancisco) [reply red.js:405]: I like this idea! It will be updated.

File: app/dojotModule/utils.js (hunk 1 above)
Comment (rodrigoaasm) [reply app/dojotModule/utils.js]: Perhaps to make the code cleaner, the use of 'spread' can be evaluated.
`const flowsElements = {  ...storageForFlows }`

**Review by rodrigoaasm — APPROVED** [review]
Ok.

**Review by Huakson — APPROVED** [review]
all right

File: app/dojotModule/utils.js (hunk 1 above)
Comment (cfrancisco) [reply app/dojotModule/utils.js]: Good idea! Spread uses shallow cloning, which works in this scenario. 
From the doc: "A shallow copy means that only the actual object gets copied. If the copied object contains nested objects — these nested objects aren’t get cloned."

**Pull request merged**